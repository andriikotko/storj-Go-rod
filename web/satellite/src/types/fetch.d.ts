// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

declare type Answer = {
    isSuccess: boolean;
    error: {
        code: any;
        message: any;
    };
};

declare type OnPageClickCallback = (index: number) => Promise<any>;

declare type CheckSelected = (index: number) => boolean;
