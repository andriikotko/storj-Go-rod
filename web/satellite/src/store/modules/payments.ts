// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/store';
import {
    AccountBalance,
    CreditCard,
    DateRange,
    PaymentsApi,
    PaymentsHistoryItem,
    PaymentsHistoryItemStatus,
    PaymentsHistoryItemType,
    ProjectUsageAndCharges,
    TokenDeposit,
} from '@/types/payments';

export const PAYMENTS_MUTATIONS = {
    SET_BALANCE: 'SET_BALANCE',
    SET_CREDIT_CARDS: 'SET_CREDIT_CARDS',
    SET_DATE: 'SET_DATE',
    CLEAR: 'CLEAR_PAYMENT_INFO',
    UPDATE_CARDS_SELECTION: 'UPDATE_CARDS_SELECTION',
    UPDATE_CARDS_DEFAULT: 'UPDATE_CARDS_DEFAULT',
    SET_PAYMENTS_HISTORY: 'SET_PAYMENTS_HISTORY',
    SET_PROJECT_USAGE_AND_CHARGES: 'SET_PROJECT_USAGE_AND_CHARGES',
    SET_CURRENT_ROLLUP_PRICE: 'SET_CURRENT_ROLLUP_PRICE',
    SET_PREVIOUS_ROLLUP_PRICE: 'SET_PREVIOUS_ROLLUP_PRICE',
    SET_PRICE_SUMMARY: 'SET_PRICE_SUMMARY',
    SET_PRICE_SUMMARY_FOR_SELECTED_PROJECT: 'SET_PRICE_SUMMARY_FOR_SELECTED_PROJECT',
    SET_PAYWALL_ENABLED_STATUS: 'SET_PAYWALL_ENABLED_STATUS',
};

export const PAYMENTS_ACTIONS = {
    GET_BALANCE: 'getBalance',
    SETUP_ACCOUNT: 'setupAccount',
    GET_CREDIT_CARDS: 'getCreditCards',
    ADD_CREDIT_CARD: 'addCreditCard',
    CLEAR_PAYMENT_INFO: 'clearPaymentInfo',
    TOGGLE_CARD_SELECTION: 'toggleCardSelection',
    CLEAR_CARDS_SELECTION: 'clearCardsSelection',
    MAKE_CARD_DEFAULT: 'makeCardDefault',
    REMOVE_CARD: 'removeCard',
    GET_PAYMENTS_HISTORY: 'getPaymentsHistory',
    MAKE_TOKEN_DEPOSIT: 'makeTokenDeposit',
    GET_PROJECT_USAGE_AND_CHARGES: 'getProjectUsageAndCharges',
    GET_PROJECT_USAGE_AND_CHARGES_CURRENT_ROLLUP: 'getProjectUsageAndChargesCurrentRollup',
    GET_PROJECT_USAGE_AND_CHARGES_PREVIOUS_ROLLUP: 'getProjectUsageAndChargesPreviousRollup',
    GET_PAYWALL_ENABLED_STATUS: 'getPaywallEnabledStatus',
};

const {
    SET_BALANCE,
    SET_CREDIT_CARDS,
    SET_DATE,
    CLEAR,
    UPDATE_CARDS_SELECTION,
    UPDATE_CARDS_DEFAULT,
    SET_PAYMENTS_HISTORY,
    SET_PROJECT_USAGE_AND_CHARGES,
    SET_PRICE_SUMMARY,
    SET_PRICE_SUMMARY_FOR_SELECTED_PROJECT,
    SET_PAYWALL_ENABLED_STATUS,
} = PAYMENTS_MUTATIONS;

const {
    GET_BALANCE,
    SETUP_ACCOUNT,
    GET_CREDIT_CARDS,
    ADD_CREDIT_CARD,
    TOGGLE_CARD_SELECTION,
    CLEAR_CARDS_SELECTION,
    CLEAR_PAYMENT_INFO,
    MAKE_CARD_DEFAULT,
    REMOVE_CARD,
    GET_PAYMENTS_HISTORY,
    MAKE_TOKEN_DEPOSIT,
    GET_PROJECT_USAGE_AND_CHARGES_CURRENT_ROLLUP,
    GET_PROJECT_USAGE_AND_CHARGES_PREVIOUS_ROLLUP,
    GET_PAYWALL_ENABLED_STATUS,
} = PAYMENTS_ACTIONS;

export class PaymentsState {
    /**
     * balance stores in cents
     */
    public balance: AccountBalance = new AccountBalance();
    public creditCards: CreditCard[] = [];
    public paymentsHistory: PaymentsHistoryItem[] = [];
    public usageAndCharges: ProjectUsageAndCharges[] = [];
    public priceSummary: number = 0;
    public priceSummaryForSelectedProject: number = 0;
    public startDate: Date = new Date();
    public endDate: Date = new Date();
    public isPaywallEnabled: boolean = true;
}

/**
 * creates payments module with all dependencies
 *
 * @param api - payments api
 */
export function makePaymentsModule(api: PaymentsApi): StoreModule<PaymentsState> {
    return {
        state: new PaymentsState(),
        mutations: {
            [SET_BALANCE](state: PaymentsState, balance: AccountBalance): void {
                state.balance = balance;
            },
            [SET_CREDIT_CARDS](state: PaymentsState, creditCards: CreditCard[]): void {
                state.creditCards = creditCards;
            },
            [SET_DATE](state: PaymentsState, dateRange: DateRange) {
                state.startDate = dateRange.startDate;
                state.endDate = dateRange.endDate;
            },
            [UPDATE_CARDS_SELECTION](state: PaymentsState, id: string | null): void {
                state.creditCards = state.creditCards.map(card => {
                    if (card.id === id) {
                        card.isSelected = !card.isSelected;

                        return card;
                    }

                    card.isSelected = false;

                    return card;
                });
            },
            [UPDATE_CARDS_DEFAULT](state: PaymentsState, id: string): void {
                state.creditCards = state.creditCards.map(card => {
                    if (card.id === id) {
                        card.isDefault = !card.isDefault;

                        return card;
                    }

                    card.isDefault = false;

                    return card;
                });
            },
            [SET_PAYMENTS_HISTORY](state: PaymentsState, paymentsHistory: PaymentsHistoryItem[]): void {
                state.paymentsHistory = paymentsHistory;
            },
            [SET_PROJECT_USAGE_AND_CHARGES](state: PaymentsState, usageAndCharges: ProjectUsageAndCharges[]): void {
                state.usageAndCharges = usageAndCharges;
            },
            [SET_PRICE_SUMMARY](state: PaymentsState, charges: ProjectUsageAndCharges[]): void {
                if (charges.length === 0) {
                    state.priceSummary = 0;

                    return;
                }

                const usageItemSummaries: number[] = charges.map(item => item.summary());

                state.priceSummary = usageItemSummaries.reduce((accumulator, current) => accumulator + current);
            },
            [SET_PRICE_SUMMARY_FOR_SELECTED_PROJECT](state: PaymentsState, selectedProjectId: string): void {
                let usageAndChargesForSelectedProject: ProjectUsageAndCharges | undefined;
                if (state.usageAndCharges.length) {
                    usageAndChargesForSelectedProject = state.usageAndCharges.find(item => item.projectId === selectedProjectId);
                }

                if (!usageAndChargesForSelectedProject) {
                    state.priceSummaryForSelectedProject = 0;

                    return;
                }

                state.priceSummaryForSelectedProject = usageAndChargesForSelectedProject.summary();
            },
            [SET_PAYWALL_ENABLED_STATUS](state: PaymentsState, isPaywallEnabled: boolean): void {
                state.isPaywallEnabled = isPaywallEnabled;
            },
            [CLEAR](state: PaymentsState) {
                state.balance = new AccountBalance();
                state.paymentsHistory = [];
                state.usageAndCharges = [];
                state.priceSummary = 0;
                state.creditCards = [];
                state.startDate = new Date();
                state.endDate = new Date();
                state.isPaywallEnabled = true;
            },
        },
        actions: {
            [GET_BALANCE]: async function({commit}: any): Promise<AccountBalance> {
                const balance: AccountBalance = await api.getBalance();

                commit(SET_BALANCE, balance);

                return balance;
            },
            [SETUP_ACCOUNT]: async function(): Promise<void> {
                await api.setupAccount();
            },
            [GET_CREDIT_CARDS]: async function({commit}: any): Promise<CreditCard[]> {
                const creditCards = await api.listCreditCards();

                commit(SET_CREDIT_CARDS, creditCards);

                return creditCards;
            },
            [ADD_CREDIT_CARD]: async function({commit}: any, token: string): Promise<void> {
                await api.addCreditCard(token);
            },
            [TOGGLE_CARD_SELECTION]: function({commit}: any, id: string): void {
                commit(UPDATE_CARDS_SELECTION, id);
            },
            [CLEAR_CARDS_SELECTION]: function({commit}: any): void {
                commit(UPDATE_CARDS_SELECTION, null);
            },
            [MAKE_CARD_DEFAULT]: async function({commit}: any, id: string): Promise<void> {
                await api.makeCreditCardDefault(id);

                commit(UPDATE_CARDS_DEFAULT, id);
            },
            [REMOVE_CARD]: async function({commit, state}: any, cardId: string): Promise<void> {
                await api.removeCreditCard(cardId);

                commit(SET_CREDIT_CARDS, state.creditCards.filter(card => card.id !== cardId));
            },
            [CLEAR_PAYMENT_INFO]: function({commit}: any): void {
                commit(CLEAR);
            },
            [GET_PAYMENTS_HISTORY]: async function({commit}: any): Promise<void> {
                const paymentsHistory: PaymentsHistoryItem[] = await api.paymentsHistory();

                commit(SET_PAYMENTS_HISTORY, paymentsHistory);
            },
            [MAKE_TOKEN_DEPOSIT]: async function({commit}: any, amount: number): Promise<TokenDeposit> {
                return await api.makeTokenDeposit(amount);
            },
            [GET_PROJECT_USAGE_AND_CHARGES_CURRENT_ROLLUP]: async function({commit, rootGetters}: any): Promise<void> {
                const now = new Date();
                const endUTC = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate(), now.getUTCHours(), now.getUTCMinutes()));
                const startUTC = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), 1, 0, 0));

                const usageAndCharges: ProjectUsageAndCharges[] = await api.projectsUsageAndCharges(startUTC, endUTC);

                commit(SET_DATE, new DateRange(startUTC, endUTC));
                commit(SET_PROJECT_USAGE_AND_CHARGES, usageAndCharges);
                commit(SET_PRICE_SUMMARY, usageAndCharges);
                commit(SET_PRICE_SUMMARY_FOR_SELECTED_PROJECT, rootGetters.selectedProject.id);
            },
            [GET_PROJECT_USAGE_AND_CHARGES_PREVIOUS_ROLLUP]: async function({commit}: any): Promise<void> {
                const now = new Date();
                const startUTC = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth() - 1, 1, 0, 0));
                const endUTC = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), 0, 23, 59, 59));

                const usageAndCharges: ProjectUsageAndCharges[] = await api.projectsUsageAndCharges(startUTC, endUTC);

                commit(SET_DATE, new DateRange(startUTC, endUTC));
                commit(SET_PROJECT_USAGE_AND_CHARGES, usageAndCharges);
                commit(SET_PRICE_SUMMARY, usageAndCharges);
            },
            [GET_PAYWALL_ENABLED_STATUS]: async function({commit, rootGetters}: any): Promise<void> {
                const isPaywallEnabled: boolean = await api.getPaywallStatus(rootGetters.user.id);

                commit(SET_PAYWALL_ENABLED_STATUS, isPaywallEnabled);
            },
        },
        getters: {
            canUserCreateFirstProject: (state: PaymentsState): boolean => {
                return state.balance.sum > 0 || state.creditCards.length > 0;
            },
            isTransactionProcessing: (state: PaymentsState): boolean => {
                return state.paymentsHistory.some((paymentsItem: PaymentsHistoryItem) => {
                    return paymentsItem.amount >= 50 && paymentsItem.type === PaymentsHistoryItemType.Transaction
                        && (paymentsItem.status === PaymentsHistoryItemStatus.Pending
                        || paymentsItem.status === PaymentsHistoryItemStatus.Paid
                        || paymentsItem.status === PaymentsHistoryItemStatus.Completed);
                }) && state.balance.sum === 0;
            },
            isBalancePositive: (state: PaymentsState): boolean => {
                return state.balance.sum > 0;
            },
        },
    };
}
