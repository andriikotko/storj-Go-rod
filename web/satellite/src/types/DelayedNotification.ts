// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { NOTIFICATION_IMAGES, NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { getId } from '@/utils/idGenerator';

export class DelayedNotification {
    private readonly successColor: string = '#DBF1D3';
    private readonly errorColor: string = '#FFD4D2';
    private readonly infoColor: string = '#D0E3FE';
    private readonly warningColor: string = '#FCF8E3';
    public readonly id: string;

    private readonly callback: Function;
    private timerId: number;
    private startTime: number;
    private remainingTime: number;

    public readonly type: string;
    public readonly message: string;
    public readonly style: any;
    public readonly imgSource: string;

    constructor(callback: Function, type: string, message: string) {
        this.callback = callback;
        this.type = type;
        this.message = message;
        this.id = getId();
        this.remainingTime = 3000;
        this.start();

        // Switch for choosing notification style depends on notification type
        switch (this.type) {
            case NOTIFICATION_TYPES.SUCCESS:
                this.style = { backgroundColor: this.successColor };
                this.imgSource = NOTIFICATION_IMAGES.SUCCESS;
                break;

            case NOTIFICATION_TYPES.ERROR:
                this.style = { backgroundColor: this.errorColor };
                this.imgSource = NOTIFICATION_IMAGES.ERROR;
                break;

            case NOTIFICATION_TYPES.WARNING:
                this.style = { backgroundColor: this.warningColor };
                this.imgSource = NOTIFICATION_IMAGES.WARNING;
                break;

            default:
                this.style = { backgroundColor: this.infoColor };
                this.imgSource = NOTIFICATION_IMAGES.NOTIFICATION;
                break;
        }
    }

    public pause(): void {
        clearTimeout(this.timerId);
        this.remainingTime -= new Date().getMilliseconds() - this.startTime;
    }

    public start(): void {
        this.startTime = new Date().getMilliseconds();
        this.timerId = setTimeout(this.callback, this.remainingTime);
    }
}
