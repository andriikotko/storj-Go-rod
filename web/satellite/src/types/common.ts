// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}
