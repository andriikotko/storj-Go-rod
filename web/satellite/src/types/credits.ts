// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all credits-related functionality.
 */
export interface CreditsApi {
    get(): Promise<CreditUsage>;
}

/**
 * CreditUsage stores information about users credits.
 * Tardigrade related logic
 */
export class CreditUsage {
    public referred: number;
    public usedCredits: number;
    public availableCredits: number;

    constructor(referred: number = 0, usedCredits: number = 0, availableCredits: number = 0) {
        this.referred = referred;
        this.usedCredits = usedCredits;
        this.availableCredits = availableCredits;
    }
}
