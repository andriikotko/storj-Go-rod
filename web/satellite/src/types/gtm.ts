// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { LoadScript } from '@/utils/loadScript';
import { MetaUtils } from '@/utils/meta';

/**
 * GoogleTagManager is a conversion tracking tag utility.
 */
export class GoogleTagManager {
    private gtmScript: LoadScript;

    /**
     * Initializes google tag manager in DOM.
     */
    public init(): void {
        const gtmID: string = MetaUtils.getMetaContent('gtm-id');
        const url = `https://www.googletagmanager.com/gtm.js?id=${gtmID}`;

        this.gtmScript = new LoadScript(url,
            () => {
                window['dataLayer'].push({
                    event: 'gtm.js',
                    'gtm.start': new Date().getTime(),
                });
            },
            (error) => {
                throw error;
            },
        );
    }

    /**
     * Removes google tag manager from DOM.
     */
    public remove(): void {
        this.gtmScript.remove();
    }
}
