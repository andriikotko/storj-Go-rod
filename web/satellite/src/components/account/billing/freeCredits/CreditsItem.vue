// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="container">
        <p class="container__item">{{ creditType }}</p>
        <p class="container__item">{{ expiration }}</p>
        <p class="container__item">{{ memoryAmount }} GB ({{ creditsItem.amount | centsToDollars }})</p>
        <p class="container__item available">{{ creditsItem.remaining | centsToDollars }}</p>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import PaymentsHistoryItemDate from '@/components/account/billing/depositAndBillingHistory/PaymentsHistoryItemDate.vue';

import { PaymentsHistoryItem, PaymentsHistoryItemType } from '@/types/payments';
import { MONTHS_NAMES } from '@/utils/constants/date';

@Component({
    components: {
        PaymentsHistoryItemDate,
    },
})
export default class CreditsItem extends Vue {
    @Prop({default: () => new PaymentsHistoryItem()})
    private readonly creditsItem: PaymentsHistoryItem;

    /**
     * Return credit type string depending on item type.
     */
    public get creditType(): string {
        const trial = 'Trial Credit';
        const referral = 'Referral Credit';

        if (this.creditsItem.type === PaymentsHistoryItemType.Coupon) {
            return trial;
        }

        return referral;
    }

    /**
     * Returns memory amount depending on item's money amount.
     */
    public get memoryAmount(): number {
        const gbPrice: number = 5.5; // in cents.

        return Math.floor(this.creditsItem.amount / gbPrice);
    }

    /**
     * Returns formatted string of expiration date.
     */
    public get expiration(): string {
        const monthNumber = this.creditsItem.end.getUTCMonth();
        const year = this.creditsItem.end.getUTCFullYear();

        return `${MONTHS_NAMES[monthNumber]} ${year}`;
    }
}
</script>

<style scoped lang="scss">
    .container {
        display: flex;
        align-items: center;
        width: 100%;

        &__item {
            min-width: 28%;
            font-family: 'font_regular', sans-serif;
            text-align: left;
            margin: 10px 0;
            font-size: 16px;
            line-height: 19px;
            color: #354049;
        }
    }

    .available {
        min-width: 16%;
        text-align: right;
    }
</style>
