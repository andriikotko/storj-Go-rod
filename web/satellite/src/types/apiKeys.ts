// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { SortDirection } from '@/types/common';

export type OnHeaderClickCallback = (sortBy: ApiKeyOrderBy, sortDirection: SortDirection) => Promise<void>;

/**
 * Exposes all apiKey-related functionality.
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKeys
     *
     * @returns ApiKey[]
     * @throws Error
     */
    get(projectId: string, cursor: ApiKeyCursor): Promise<ApiKeysPage>;

    /**
     * Create new apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    create(projectId: string, name: string): Promise<ApiKey>;

    /**
     * Delete existing apiKey
     *
     * @returns null
     * @throws Error
     */
    delete(ids: string[]): Promise<void>;
}

/**
 * Holds api keys sorting parameters.
 */
export enum ApiKeyOrderBy {
    NAME = 1,
    CREATED_AT,
}

/**
 * ApiKeyCursor is a type, used to describe paged api keys list.
  */
export class ApiKeyCursor {
    public constructor(
        public search: string = '',
        public limit: number = 6,
        public page: number = 1,
        public order: ApiKeyOrderBy = ApiKeyOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
    ) {}
}

/**
 * ApiKeysPage is a type, used to describe paged api keys list.
 */
export class ApiKeysPage {
    public constructor(
        public apiKeys: ApiKey[] = [],
        public search: string = '',
        public order: ApiKeyOrderBy = ApiKeyOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
        public limit: number = 6,
        public pageCount: number = 0,
        public currentPage: number = 1,
        public totalCount: number = 0,
    ) {}
}

/**
 * ApiKey class holds info for ApiKeys entity.
 */
export class ApiKey {
    public isSelected: boolean;

    constructor(
        public id: string = '',
        public name: string = '',
        public createdAt: Date = new Date(),
        public secret: string = '',
    ) {
        this.isSelected = false;
    }

    /**
     * Returns created date as a local date string.
     */
    public localDate(): string {
        return this.createdAt.toLocaleDateString();
    }
}
