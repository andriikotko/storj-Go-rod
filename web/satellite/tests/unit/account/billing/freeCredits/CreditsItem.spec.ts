// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import CreditsItem from '@/components/account/billing/freeCredits/CreditsItem.vue';

import { PaymentsHistoryItem, PaymentsHistoryItemType } from '@/types/payments';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const coupon = new PaymentsHistoryItem('testId', 'desc', 275, 0, 'Active', '', new Date(1), new Date(1), PaymentsHistoryItemType.Coupon, 275);

localVue.filter('centsToDollars', (cents: number): string => {
    return `$${(cents / 100).toFixed(2)}`;
});

describe('CreditsItem', (): void => {
    it('renders correctly', (): void => {
        const wrapper = mount(CreditsItem, {
            localVue,
            propsData: {
                creditsItem: coupon,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
