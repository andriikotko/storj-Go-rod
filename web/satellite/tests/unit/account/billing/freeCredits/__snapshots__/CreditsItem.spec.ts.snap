// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsItem renders correctly 1`] = `
<div class="container">
  <p class="container__item">Trial Credit</p>
  <p class="container__item">January 1970</p>
  <p class="container__item">50 GB ($2.75)</p>
  <p class="container__item available">$2.75</p>
</div>
`;
